<launch>
  <!-- Set the eye mode (see README for more info) -->
  <arg name="camera_parent_frame"/>
  <arg name="marker_parent_frame"/>


  <!-- The camera topics and frame to use for calibration -->
  <arg name="camera" doc="camera namespace"/>
  <arg name="image"                 default="$(arg camera)/image_rect_color"/>
  <arg name="camera_info"           default="$(arg camera)/camera_info"/>
  <arg name="camera_frame"          default="$(arg camera)_optical_frame"/>
  <arg name="marker_frame"          default="/hand_eye/$(arg camera)/aruco_marker_frame"/>

  <!-- leave empty and the pose will be published wrt param parent_name -->
  <arg name="reference_frame"       default="$(arg camera_frame)"/>

  <!-- Publish frames while calibrating -->
  <arg name="publish_tf" default="true"/>

  <!-- Ask to commit after each sample -->
  <arg name="interactive" default="true"/>

  <!-- If the connector node should be spawned in a new window -->
  <arg name="separate_window" default="true"/>
  <arg unless="$(arg separate_window)" name="LAUNCH_PREFIX" value=""/>
  <arg if="$(arg separate_window)" name="LAUNCH_PREFIX" value="xterm -e"/>

  <!-- Transform from the camera base link to the optical link (only necessary
       if you want aruco_hand_eye to compute the transform for you) -->
  <arg name="xyz_optical_base" default="[0.0, 0.0, 0.0]"/>
  <arg name="rpy_optical_base" default="[0.0, 0.0, 0.0]"/>

  <group ns="$(arg camera)">

    <!-- Run the VISP solver to determine the extrinsic parameters -->
    <node name="hand_eye_solver"
      pkg="visp_hand2eye_calibration"
      type="visp_hand2eye_calibration_calibrator">
      <remap from="/compute_effector_camera_quick" to="compute_effector_camera_quick"/>
    </node>

    <node pkg="tf" type="static_transform_publisher" name="object_startframe_transform" args="-0.280, 0, -0.025 1.57079632679, 0, 1.57079632679 QR_object start_frame 20" /> 

    <!-- Connect the visp auto tracker to the solver -->
    <node name="kinect_bhand_calib"
      launch-prefix="$(arg LAUNCH_PREFIX)"
      pkg="kinect_bhand_calib"
      type="calibrate.py"
      output="screen">
      <param name="sample_rate" value="4"/>
      <param name="camera_parent_frame" value="$(arg camera_parent_frame)"/>
      <param name="marker_parent_frame" value="$(arg marker_parent_frame)"/>
      <param name="publish_tf" value="$(arg publish_tf)"/>
      <param name="tf_suffix" value=""/>
      <param name="interactive" value="$(arg interactive)"/>
      <rosparam param="xyz_optical_base" subst_value="true">$(arg xyz_optical_base)</rosparam>
      <rosparam param="rpy_optical_base" subst_value="true">$(arg rpy_optical_base)</rosparam>
    </node>


  </group>

</launch>
